### PR TITLE
Update Nextjs to 13.4.12

### PR DESCRIPTION
**Pull request by danielcliu-ifixit**
## Overview

[As recommended to try to fix this error](https://github.com/iFixit/react-commerce/pull/1846#issuecomment-1649423093) lets update next js to the lastest version. This just upgrades patches so everything should be backwards compatible, meaning no code changes should be necessary on our end.

## QA
Does everything still seem fine?

CC @federicobadini @danielbeardsley @sterlinghirsh 
Connects: https://github.com/iFixit/react-commerce/issues/1808
Closes: #1852

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -726,7 +730,7 @@ importers:
         version: link:../helpers
       '@sentry/nextjs':
         specifier: ^7.53.1
-        version: 7.53.1(next@13.4.1)(react@18.2.0)(webpack@5.73.0)
+        version: 7.53.1(next@13.4.1)(react@18.2.0)
```
Comment (danielcliu-ifixit) [inline pnpm-lock.yaml:733]: does this seem right?

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -1,5 +1,9 @@
 lockfileVersion: '6.0'
 
+settings:
```
Comment (danielcliu-ifixit) [inline pnpm-lock.yaml:3]: do we want these changes?

File: pnpm-lock.yaml [hunk 3]
```diff
@@ -1003,13 +1007,15 @@ packages:
 
   /@aws-crypto/ie11-detection@3.0.0:
     resolution: {integrity: sha512-341lBBkiY1DfDNKai/wXM3aujNBkXR7tq1URPQDL9wi3AUbI80NR74uF1TXHMm7po1AcnFk8iu2S2IeU/+/A+Q==}
+    requiresBuild: true
```
Comment (danielcliu-ifixit) [inline pnpm-lock.yaml:1010]: do we want these changes?

**Comment by danielcliu-ifixit** [comment]
deploy_block :v:  on me restructuring the commits and such

**Comment by federicobadini** [comment]
CR 👍

**Comment by danielcliu-ifixit** [comment]
sorry about the commit mess.

Looked over the changes for `@sentry/nextjs` didn't see anything worrying but we should keep an eye on the error logs when this deploys

**Comment by danielcliu-ifixit** [comment]
forgot to un_deploy_block :v:

**Comment by danielbeardsley** [comment]
> By the way this should also fix this: [vercel/next.js#49789](https://github.com/vercel/next.js/issues/49789).

>> which prevents any client-side code from running.

Ooof. 

The Safari 12-13 version range accounts for ~0.5% of our traffic and revenue. Google Analytics doesn't show the revenue dropping off a cliff, so I'm not sure we were affected.

**Comment by danielcliu-ifixit** [comment]
QA :cactus: carry-over

**Review by aburke07 — APPROVED** [review]
CR :alien:

**Pull request merged**